contracts/upgradable/FinalProxy.sol

```diff
@@ -79,7 +80,11 @@ contract FinalProxy is Proxy, IFinalProxy {
         }
         if (msg.sender != owner) revert NotOwner();
 
+        bytes32 id = contractId();
         finalImplementation_ = Create3.deploy(FINAL_IMPLEMENTATION_SALT, bytecode);
+
+        if (id != bytes32(0) && IUpgradable(finalImplementation_).contractId() != id) revert InvalidImplementation();
```

Comment:
Needs a comment around why the exception for the empty case

Reply:
Added